lib/samples/edit_with_branch_versioning/edit_with_branch_versioning.dart

```diff
@@ -0,0 +1,613 @@
+// Copyright 2025 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import 'package:arcgis_maps/arcgis_maps.dart';
+import 'package:arcgis_maps_sdk_flutter_samples/common/common.dart';
+import 'package:flutter/material.dart';
+
+// ViewModel Class
+class EditWithBranchVersioningModel extends ChangeNotifier {
+  EditWithBranchVersioningModel() {
+    // Initialize the current version name.
+    currentVersionNameNotifier.value = serviceGeodatabase.defaultVersionName;
+  }
+
+  // The names of the versions added by the user.
+  //
+  // - Note: To get a full list of versions, use `ServiceGeodatabase.versions`.
+  // In this sample, only the default version and versions created in current session are shown.
+  final existingVersionNames = <String>[];
+
+  // Initially centers the map's viewpoint on Naperville, IL, USA.
+  final map = ArcGISMap.withBasemapStyle(BasemapStyle.arcGISStreets)
+    ..initialViewpoint = Viewpoint.fromCenter(
+      ArcGISPoint(
+        x: -9811970,
+        y: 5127180,
+        spatialReference: SpatialReference.webMercator,
+      ),
+      scale: 4000,
+    );
+
+// A geodatabase connected to the damage assessment feature service.
+  final serviceGeodatabase = ServiceGeodatabase.withUri(
+    Uri.parse(
+      'https://sampleserver7.arcgisonline.com/server/rest/services/DamageAssessment/FeatureServer',
+    ),
+  );
+
+  // Use a ValueNotifier to track the current version name.
+  final currentVersionNameNotifier = ValueNotifier<String>('');
+
+  // Update the current version name and notify listeners
+  void updateCurrentVersionName() {
+    currentVersionNameNotifier.value = serviceGeodatabase.versionName;
+    notifyListeners();
+  }
+
+  late FeatureLayer featureLayer;
+  Feature? selectedFeature;
+
+// A Boolean value indicating whether the geodatabase's current version it's default version.
```

Comment:
`it's` should just be `its`, and there's a missing "is" in the sentence. Also I wouldn't capitalize Boolean.
"A boolean value indicating whether the geodatabase's current version is its default version."